compiler/dyno/lib/parsing/parsing-queries.cpp

```diff
@@ -341,6 +341,31 @@ bool functionWithIdHasWhere(Context* context, ID id) {
   return functionWithIdHasWhereQuery(context, id);
 }
 
+void setConfigParams(Context* context, std::vector<std::pair<std::string,std::string>> keys) {
+  QUERY_STORE_INPUT_RESULT(configParams, context, keys);
+}
+
+const
+std::vector<std::pair<std::string,std::string>>& configParams(Context* context) {
+  QUERY_BEGIN_INPUT(configParams, context);
+
+  // return empty configParams if not already set using setConfigParams
+  std::vector<std::pair<std::string, std::string>> result;
+
+  return QUERY_END(result);
+}
+
+void
+useConfigParam(Context* context, std::string name, ID id) {
+  QUERY_STORE_INPUT_RESULT(nameToConfigParamId, context, id, name);
+}
+
+// check if config was used already
+const ID& nameToConfigParamId(Context* context, std::string name) {
+  QUERY_BEGIN_INPUT(nameToConfigParamId, context, name);
+  ID result;
```

Comment:
Need a comment here like `return empty ID if ID not already set using useConfigParam`